components/tidb_query_datatype/src/codec/mysql/enums.rs

```diff
@@ -0,0 +1,71 @@
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Enum {
+    data: Vec<u8>,
+    offset: Vec<usize>,
+    value: usize,
+}
+
+impl Enum {
+    fn get(&self, idx: usize) -> &[u8] {
+        assert!(idx < self.offset.len());
+
+        let start = self.offset[idx];
+        let end = if idx < self.offset.len() - 1 {
+            self.offset[idx + 1]
+        } else {
+            self.offset.len()
+        };
+
+        &self.data[start..end]
+    }
+}
+
+impl ToString for Enum {
+    fn to_string(&self) -> String {
+        let buf = Vec::from(self.get(self.value));
+
+        unsafe { String::from_utf8_unchecked(buf) }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialOrd, PartialEq, Ord, Eq)]
```

Comment:
What about only compare value instead of deriving the default `Ord`? By the way, is it required to implement `Ord` and `Eq`?

Reply:
- Both `ScalarValue` and `VectorValue` need to implement `PartialEq`
- `impl<'a> PartialOrd for ScalarValueRef<'a> ` requires implement `cmp`

Reply:
I will implement `Ord` via using value instead of deriving here.